Mail/Views/New Message/ComposeEditor/ComposeEditor.swift

```diff
@@ -0,0 +1,157 @@
+
+    @EnvironmentObject private var attachmentsManager: AttachmentsManager
+
+    @State private var toolbar = EditorToolbarView()
+    @StateObject private var textAttributes = TextAttributes()
+
+    @ModalState(context: ContextKeys.compose) private var isShowingLinkAlert = false
+    @ModalState(context: ContextKeys.compose) private var isShowingFileSelection = false
+    @ModalState(context: ContextKeys.compose) private var isShowingPhotoLibrary = false
+    @ModalState(context: ContextKeys.compose) private var isShowingCamera = false
+
+    @FocusState var focusedField: ComposeViewFieldType?
+
+    @ObservedRealmObject var draft: Draft
+
+    @Binding var isShowingAI: Bool
+
+    let messageReply: MessageReply?
+
+    private var isRemoteContentBlocked: Bool {
+        return UserDefaults.shared.displayExternalContent == .askMe && messageReply?.frozenMessage.localSafeDisplay == false
+    }
+
+    var body: some View {
+        RichHTMLEditor(html: $draft.body, textAttributes: textAttributes)
+            .focused($focusedField, equals: .editor)
+            .onAppear(perform: setupToolbar)
+            .editorInputAccessoryView(toolbar)
+            .editorCSS(Self.customCSS)
+            .introspectEditor(perform: setupEditor)
+            .onJavaScriptFunctionFail(perform: reportJavaScriptError)
+            .customAlert(isPresented: $isShowingLinkAlert) {
+                AddLinkView(actionHandler: didCreateLink)
+            }
+            .sheet(isPresented: $isShowingFileSelection) {
+                DocumentPicker(pickerType: .selectContent([.item], didPickDocument))
+                    .ignoresSafeArea()
+            }
+            .sheet(isPresented: $isShowingPhotoLibrary) {
+                ImagePicker(completion: didPickImage)
+                    .ignoresSafeArea()
+            }
+            .fullScreenCover(isPresented: $isShowingCamera) {
+                CameraPicker(completion: didTakePhoto)
+                    .ignoresSafeArea()
+            }
+    }
+
+    private func setupEditor(_ editor: RichHTMLEditorView) {
+        Task {
+            let contentBlocker = ContentBlocker(webView: editor.webView)
+            try? await contentBlocker.setRemoteContentBlocked(isRemoteContentBlocked)
+        }
+    }
+
+    private func setupToolbar() {
+        toolbar.setTextAttributes(textAttributes)
+        toolbar.mainButtonItemsHandler = didTapMainToolbarButton
+    }
+
+    private func didTapMainToolbarButton(_ action: EditorToolbarAction) {
+        switch action {
+        case .link:
+            isShowingLinkAlert = true
+        case .ai:
+            isShowingAI = true
+        case .addFile:
+            isShowingFileSelection = true
+        case .addPhoto:
+            isShowingPhotoLibrary = true
+        case .takePhoto:
+            isShowingCamera = true
+        case .programMessage:
+            showWorkInProgressSnackBar()
+        default:
+            print("Action not handled.")
```

Comment:
Maybe use DDLogWarn instead of print or break

Reply:
Done 👍